v0/destinations/marketo_static_list/util.js

```diff
@@ -0,0 +1,210 @@
+      },
+      marketoResponse,
+      undefined, // represents authErrorCategory
+      DESTINATION
+    );
+  } else if (errors && MARKETO_THROTTLED_CODES.indexOf(errors[0].code) > -1) {
+    throw new ApiError(
+      `Request Failed for Marketo Static List, ${errors[0].message} (Throttled).${sourceMessage}`,
+      429,
+      {
+        scope: TRANSFORMER_METRIC.MEASUREMENT_TYPE.API.SCOPE,
+        meta: TRANSFORMER_METRIC.MEASUREMENT_TYPE.API.META.THROTTLED
+      },
+      marketoResponse,
+      undefined,
+      DESTINATION
+    );
+  } else if (errors && MARKETO_RETRYABLE_CODES.indexOf(errors[0].code) > -1) {
+    throw new ApiError(
+      `Request Failed for Marketo Static List, ${errors[0].message} (Retryable).${sourceMessage}`,
+      500,
+      {
+        scope: TRANSFORMER_METRIC.MEASUREMENT_TYPE.API.SCOPE,
+        meta: TRANSFORMER_METRIC.MEASUREMENT_TYPE.API.META.RETRYABLE
+      },
+      marketoResponse,
+      undefined,
+      DESTINATION
+    );
+  }
+};
+
+const marketoResponseHandler = (
+  destResponse,
+  sourceMessage,
+  stage,
+  rudderJobMetadata,
+  authCache
+) => {
+  const { status, response } = destResponse;
+  // if the responsee from destination is not a success case build an explicit error
+  if (!isHttpStatusSuccess(status)) {
+    throw new ApiError(
+      `[Marketo Static List Response Handler] - Request failed  with status: ${status}`,
+      status,
+      {
+        scope: TRANSFORMER_METRIC.MEASUREMENT_TYPE.API.SCOPE,
+        meta: getDynamicMeta(status)
+      },
+      destResponse,
+      undefined,
+      DESTINATION
+    );
+  }
+  if (isHttpStatusSuccess(status)) {
+    // for authentication requests
+    if (response && response.access_token && response.expires_in) {
+      return response;
+    }
+    // marketo application level success
+    if (response && response.success) {
+      return response;
+    }
+    // marketo application level failure
+    if (response && !response.success) {
+      // checking for invalid/expired token errors and evicting cache in that case
+      // rudderJobMetadata contains some destination info which is being used to evict the cache
+      if (response.errors && rudderJobMetadata?.destInfo) {
+        const { authKey } = rudderJobMetadata.destInfo;
+        if (
+          authCache &&
+          authKey &&
+          response.errors.some(errorObj => {
+            return errorObj.code === "601" || errorObj.code === "602";
+          })
+        ) {
+          authCache.del(authKey);
+        }
+      }
+      marketoApplicationErrorHandler(destResponse, sourceMessage);
+    }
+  }
+  // More readable error message
+  let message = `Error occurred ${sourceMessage}`;
+  if (response.errors.length > 0 && response.errors[0].message) {
+    message += ` -> ${response.errors[0].message}`;
+  }
+  // Marketo sent us some failure which is not handled
+  throw new ApiError(message, 400, null, destResponse);
+};
```

Comment:
Lots of code duplication it seems for marketo, and marketo audience upload can we refer this from common marketo util file?

Reply:
Yeah, Saw the similarities and already working on that

Reply:
Can I make some changes in Marketo utils to make some functions more general like passing destination Name as a parameter so that it could be used in while throwing error?

Reply:
Yes